app/shared/components/Tools/Table/Row/Wallet.js

```diff
@@ -43,12 +42,18 @@ class ToolsTableRowWallet extends Component<Props> {
     });
   };
   removeWallet = (account, authorization) => {
-    const { actions, settings, walletCount } = this.props;
+    const { actions, settings, wallets } = this.props;
 
     actions.removeWallet(settings.chainId, account, authorization);
 
-    if (walletCount === 1) {
-      actions.changeModule('');
+    if (wallets.length === 1) {
+      return actions.changeModule('');
+    }
+
+    if (settings.account === account) {
+      const otherWallet = wallets.find(wallet => wallet.account !== account);
+
+      actions.useWallet(settings.chainId, otherWallet.account, authorization);
```

Comment:
Should the `authorization` piece here be `otherWallet.authorization`?

Reply:
Yes it should! Nice catch!